storage/store/collections.go

```diff
@@ -93,16 +82,40 @@ func (c *Collections) LightByID(colID flow.Identifier) (*flow.LightCollection, e
 		return nil, fmt.Errorf("could not retrieve collection: %w", err)
 	}
 
+	if err != nil {
+		return nil, err
+	}
+
 	return &collection, nil
 }
 
 // Remove removes a collection from the database.
 // Remove does not error if the collection does not exist
 // any error returned are exceptions
 func (c *Collections) Remove(colID flow.Identifier) error {
-	err := c.db.WithReaderBatchWriter(func(rw storage.ReaderBatchWriter) error {
+	col, err := c.LightByID(colID)
+	if err != nil {
+		if errors.Is(err, storage.ErrNotFound) {
+			// already removed
+			return nil
+		}
+
+		return fmt.Errorf("could not retrieve collection: %w", err)
+	}
+
+	err = c.db.WithReaderBatchWriter(func(rw storage.ReaderBatchWriter) error {
+		// remove transaction indices
+		for _, txID := range col.Transactions {
+			err = operation.RemoveCollectionTransactionIndices(rw.Writer(), txID)
```

Comment:
`Collections.Store` above inserts transactions. Should `Remove` also remove the transactions?